pkg/common-controller/snapshot_controller.go

```diff
@@ -715,6 +715,9 @@ func (ctrl *csiSnapshotCommonController) createSnapshotContent(snapshot *crdv1.V
 		klog.Errorf("failed to update content store %v", err)
 	}
 
+	msg = fmt.Sprintf("Snapshot %s was successfully created by the CSI driver.", utils.SnapshotKey(snapshot))
+	ctrl.eventRecorder.Event(snapshot, v1.EventTypeNormal, "SnapshotCreated", msg)
+
```

Comment:
At this point, the VolumeSnapshotContent is just created.  The CSI snapshotter sidecar will call the CSI driver to create a snapshot, but we don't know whether it is completed yet.

In updateSnapshotStatus (https://github.com/kubernetes-csi/external-snapshotter/blob/v4.1.1/pkg/common-controller/snapshot_controller.go#L1072), we know whether snapshot is created successfully.

Add a snapshot is created successfully event in this if block:
https://github.com/kubernetes-csi/external-snapshotter/blob/v4.1.1/pkg/common-controller/snapshot_controller.go#L1155

Add a snapshot is ready to use event in this if block:
https://github.com/kubernetes-csi/external-snapshotter/blob/v4.1.1/pkg/common-controller/snapshot_controller.go#L1162

Reply:
Thank you for the detailed explaination, Xing!